### ISSUE TITLE
The first inference timing for pure PyTorch is unexpectedly fast and should be ignored

### ISSUE THREAD

**Issue by siahuat0727** [huggingface/transformers#13944]
## Environment info

- `transformers` version: 4.11.3
- Platform: Linux-5.4.0-1056-azure-x86_64-with-glibc2.10
- Python version: 3.8.5
- PyTorch version (GPU?): 1.9.1+cu102 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?:

## Information

Hi, thanks for the benchmark library, which has helped me simplify a lot in my project.

But I ran into a strange issue that the inference speed slows down when using `torchscript=True`. After digging into this issue, I found that the first timing for `torchscript=False` is strangely fast. (see below section for the details)
I think it is better to perform warmup for all situations instead of only:
https://github.com/huggingface/transformers/blob/91758e399f8c4bf81820a8af6a257682ccea0223/src/transformers/benchmark/benchmark.py#L198

I will be happy to send a PR if this suggestion is accepted.

## To reproduce

```python
# test.py
from transformers import PyTorchBenchmark, PyTorchBenchmarkArguments

args = PyTorchBenchmarkArguments(models=["bert-base-uncased"], batch_sizes=[8], sequence_lengths=[512], torchscript=False, env_print=True)
benchmark = PyTorchBenchmark(args)
benchmark.run()
```
1. Add `print(runtimes)` at L213

https://github.com/huggingface/transformers/blob/91758e399f8c4bf81820a8af6a257682ccea0223/src/transformers/benchmark/benchmark.py#L208-L212

2. Run `test.py` with `torchscript=False`

```
[1.2410342500079423, 1.7289500490296632, 1.7288817439693958, 1.7287787408567965, 1.7287089368328452]
====================       INFERENCE - SPEED - RESULT       ====================
--------------------------------------------------------------------------------
          Model Name             Batch Size     Seq Length     Time in s   
--------------------------------------------------------------------------------
      bert-base-uncased              8              512            0.124     
--------------------------------------------------------------------------------
...
```

3. Run `test.py` with `torchscript=True`

```
[1.6381433540955186, 1.6382532438728958, 1.6387098210398108, 1.6382117359898984, 1.6384075228124857]

====================       INFERENCE - SPEED - RESULT       ====================
--------------------------------------------------------------------------------
          Model Name             Batch Size     Seq Length     Time in s   
--------------------------------------------------------------------------------
      bert-base-uncased              8              512            0.164     
--------------------------------------------------------------------------------

...

====================        ENVIRONMENT INFORMATION         ====================
- transformers_version: 4.11.3
- framework: PyTorch
- use_torchscript: True
- framework_version: 1.9.1+cu102
- python_version: 3.8.5
- system: Linux
- cpu: x86_64
- architecture: 64bit
- date: 2021-10-09
- time: 14:50:09.526534
- fp16: False
- use_multiprocessing: True
- only_pretrain_model: False
- cpu_ram_mb: N/A
- use_gpu: True
- num_gpus: 1
- gpu: N/A
- gpu_ram_mb: N/A
- gpu_power_watts: N/A
- gpu_performance_state: N/A
- use_tpu: False
```

## Expected behavior

Perform warmup for all situations to avoid getting unexpected inference speed.

**Comment by siahuat0727** [comment]
It would be nice if someone could point out the reason for this strange behavior. :)

**Comment by siahuat0727** [comment]
@patrickvonplaten 
Hello, I found from the guide that you are the right people for the benchmark library. :smile:
Do you think this suggestion should be accepted?

**Comment by patrickvonplaten** [comment]
Hey @siahuat0727,

Thanks for your issue. We don't actively maintain the benchmarking tools of `transformers` anymore and also recommend to not use it as it's quite outdated and has not been shown to be very accurate.
